src/coreclr/jit/emitarm.cpp

```diff
@@ -6790,7 +6790,7 @@ void emitter::emitDispImm(int imm, bool addComma, bool alwaysHex /* =false */)
         printf("%d", imm);
     else if ((imm > 0) ||
              (imm == -imm) || // -0x80000000 == 0x80000000. So we don't want to add an extra "-" at the beginning.
-             (emitComp->opts.disDiffable && (imm == 0xD1FFAB1E))) // Don't display this as negative
+             (emitComp->opts.disDiffable && (imm == 0xD1FFAB1EL))) // Don't display this as negative
```

Comment:
Sorry I didn't notice this earlier, but I think this needs to be:
```suggestion
             (emitComp->opts.disDiffable && (imm == (int)0xD1FFAB1E))) // Don't display this as negative
```

I'm surprised that GCC doesn't warn here, isn't this comparison always false?

Reply:
Yes, it seems clang and gcc have different sensitivity/tolerance levels for that value that has just overflown the type limit https://godbolt.org/z/KPof67PY3 😄